docs-src/cloud/limits.md

```diff
@@ -169,7 +169,42 @@ These operations create Events that are then logged in the Event History.
 The transaction size limit restricts the total size of Events that can be accommodated within a single transaction.
 
 The size limit for any given [Event History](/concepts/what-is-an-event-history) transaction is 4 MB.
-This isn't configurable.
+This limit is non-configurable for Temporal Cloud.
+
+**What is the blob size limit for a transaction?**
+
+- **Blob size limit** for Payloads, including Workflow context and each Workflow and Activity argument and return value:
+  - Temporal warns at 256 KB: `Blob size exceeds limit`.
+  - Temporal errors at 2 MB: `ErrBlobSizeExceedsLimit: Blob data size exceeds limit.`
+
+This limit is non-configurable for Temporal Cloud.
+
+The following provides solutions for handling large payloads.
+
+**Solutions for handling large payloads**
+
+If you're at risk of reaching the blob size limit, consider the following solutions:
+
+1. Use compression with a [custom Data Converter](/concepts/what-is-a-custom-data-converter) for large payloads.
+2. Break larger batches of commands into smaller batch sizes.
+
+**Monitoring and Debugging**
+
+To proactively identify and address issues related to blob size limits, consider the following:
+
+1. Monitor the `temporal_request_failure` SDK metric for operations such as:
+   - `RespondWorkflowTaskCompleted`
+   - `ActivityTaskCompleted`
+2. Set up alerts on the `temporal_request_failure` SDK metric for these operations to receive notifications when failures occur.
+3. Implement a gRPC proxy between the Client Worker and the service to obtain gRPC message size information. This allows you to check the message size and identify potential issues before they cause failures.
+
+**Versioning Workflows**
```

Comment:
This subsection feels out of place on the present page. It's true that one might need to modify their workflow code to fix the above issue, and might need to use versioning to preserve compatibility, but there are tons of other cases where they might need to do so too, and we're not discussing Workflow Versioning every single time.